### PR TITLE
Issue#1043 notification

### PR DESCRIPTION
**Pull request by ioaNikas**
Removed deprecated or unused code.
Redefined notification model to be usable by front and back end of the application (issue #1043).

### CODE REVIEW

File: apps/backend-functions/src/delivery.ts [hunk 1]
```diff
@@ -90,39 +100,39 @@ async function notifyOnNewSignee(delivery: any, organizations: OrganizationDocum
...
-  const newStakeholderOrg = await getDocument<OrganizationDocument>(`orgs/${newStakeholder!.id}`);
+  const newStakeholderOrg= await getDocument<OrganizationDocument>(`orgs/${newStakeholder!.id}`);
 
   if (!newStakeholderOrg) {
     throw new Error('This organization doesn\'t exist !');
   }
 
-  const notifications = createSignatureNotifications(organizations, movie, delivery, newStakeholderOrg);
+  const snapObject: SnapObject = {
+    organization: {id: newStakeholderOrg.id, name: newStakeholderOrg.name} as PublicOrganization,
```
Comment (lsenta) [inline apps/backend-functions/src/delivery.ts]: bad smell: only use `as` for things that comes from the database. Your types have a problem here.

File: apps/backend-functions/src/delivery.ts [hunk 2]
```diff
@@ -90,39 +100,39 @@ async function notifyOnNewSignee(delivery: any, organizations: OrganizationDocum
     throw new Error('This stakeholder doesn\'t exist !');
   }
 
-  const newStakeholderOrg = await getDocument<OrganizationDocument>(`orgs/${newStakeholder!.id}`);
+  const newStakeholderOrg= await getDocument<OrganizationDocument>(`orgs/${newStakeholder!.id}`);
```
Comment (lsenta) [inline apps/backend-functions/src/delivery.ts]: spacing, autoformat & re-read your PR ;)

File: apps/backend-functions/src/delivery.ts [hunk 3]
```diff
@@ -90,39 +100,39 @@ async function notifyOnNewSignee(delivery: any, organizations: OrganizationDocum
...
- * Create custom notifications for deliveries signatures. If it is used for a non-final signature,
- * pass the signatory's Organization as the 4th argument to get the correct message displayed in notification.
- */
+/**  Create custom notifications for deliveries signatures. */
 function createSignatureNotifications(
-  organizations: OrganizationDocument[],
-  movie: Movie,
-  delivery: Delivery,
-  newStakeholderOrg?: OrganizationDocument
+  organizationsOfDocument: OrganizationDocument[],
```
Comment (lsenta) [inline apps/backend-functions/src/delivery.ts]: `organizationDocuments` the current name looks like something that we use to describe how documents are organized.

File: apps/backend-functions/src/invitation.ts [hunk 4]
```diff
@@ -132,6 +134,7 @@ async function onDocumentInvitationAccept(invitation: InvitationToWorkOnDocument
   const stakeholderId = invitation.organization.id;
   const docId = invitation.docId;
   const delivery = await getDocument<Delivery>(`deliveries/${docId}`);
+  const movie = await getDocument<Movie>(`movies/${delivery.movieId}`);
```
Comment (lsenta) [inline apps/backend-functions/src/invitation.ts]: You went too far, the PR changes the data.

File: libs/notification/src/lib/notification/+state/notification.firestore.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  docId: string;
+  type: NotificationType;
+  movie: PublicMovie;
+  organization?: PublicOrganization;
+}
+
+/** Generic informations for a Notification. */
+export interface NotificationDocument extends NotificationOptions {
+  id: string;
+  app: string;
```
Comment (lsenta) [inline libs/notification/src/lib/notification/+state/notification.firestore.ts]: You went toooooo far! This changes the model! (appIcon -> app)

File: libs/notification/src/lib/notification/notification-item/notification-item.component.ts [hunk 6]
```diff
@@ -11,31 +12,45 @@ import { NotificationService, Notification } from '../+state';
...
-      this.service.readNotification(this.notification.id);
-    } catch (error) {
-      console.error(error);
-      throw new Error(this.errorMessage);
+  /** Creates a message based on the notification.type. */
+  public get message(): string {
+
+    const movieName = this.notification.movie.title.original;
+
+    switch (this.notification.type) {
```
Comment (lsenta) [inline libs/notification/src/lib/notification/notification-item/notification-item.component.ts]: Nice

File: libs/notification/src/lib/notification/+state/notification.firestore.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+  isRead: boolean;
+  date: FirebaseFirestore.FieldValue;
+};
+
+/** Type of Notification depending of its origin. */
+export const enum NotificationType {
```
Comment (lsenta) [inline libs/notification/src/lib/notification/+state/notification.firestore.ts]: Move this code above,
try to keep your code ordered: I need to know `NotificationType` BEFORE `NotificationOptions`

File: libs/notification/src/lib/notification/+state/notification.firestore.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  pathToDocument = 'pathToDocument'
+}
+
+/** Createa a Notification with required and generic informations. */
+export function createNotification(notification: NotificationOptions): NotificationDocument {
+  return {
+    id: admin.firestore().collection('notifications').doc().id,
+    isRead: false,
+    date: admin.firestore.FieldValue.serverTimestamp(),
+    app: 'delivery',
```
Comment (lsenta) [inline libs/notification/src/lib/notification/+state/notification.firestore.ts]: Could you use an enum somewhere, I bet there's an `AppSomething` somewhere.

File: libs/notification/src/lib/notification/+state/notification.firestore.ts (hunk 5 above)
Comment (ioaNikas) [reply libs/notification/src/lib/notification/+state/notification.firestore.ts]: Indeed, I will keep a trace of this for @Mussuranae who's working on data model right now. Won't happen again.

File: libs/notification/src/lib/notification/+state/notification.firestore.ts (hunk 8 above)
Comment (ioaNikas) [reply libs/notification/src/lib/notification/+state/notification.firestore.ts]: I'm using it now and putting the app enum in static models as it is used in a lot of places. Btw, static models is in movie but his content is used by other libs than movie. We maybe should think about moving it elsewhere.

**Pull request merged**